src/client/mapblock_mesh.cpp

```diff
@@ -702,6 +702,16 @@ MapBlockMesh::MapBlockMesh(Client *client, MeshMakeData *data, v3s16 camera_offs
 				tex.MinFilter = video::ETMINF_NEAREST_MIPMAP_NEAREST;
 				tex.MagFilter = video::ETMAGF_NEAREST;
 			});
+			/*
+			 * The second layer is for overlays, but uses the same vertex positions
+			 * as the first, which quickly leads to z-fighting.
+			 * To fix this we can offset the polygons in the direction of the camera.
+			 * This only affects the depth buffer and leads to no visual gaps in geometry.
+			 */
+			if (layer == 1) {
```

Comment:
Suggestion for `tile.h`:
```C++
enum {
	TILE_LAYER_NORMAL = 0,
	TILE_LAYER_OVERLAY = 1,
	MAX_TILE_LAYERS = 2
};
```
And here (and in `createSpecialNodeMesh`): `if (layer == TILE_LAYER_OVERLAY)`

1. Easier code searching by keywords
2. Mostly self-explaining code branches

Reply:
I don't disagree, but `0` and `1` are already used as constants all over the code and having an enum value used only once may be confusing than not having it.
I'll add an explanation next to `MAX_TILE_LAYERS` however.

Reply:
Alright. Thank you.